index.js

```diff
@@ -83,6 +93,106 @@ function attachToBody (options, req, reply, next) {
   })
 }
 
+function validateSerializedKey (tokens, value) {
```

Comment:
hard to follow what's going on here but I guess this is something that we have to do. I have a feeling that this code may be coming from somewhere else, possibly slightly modified though. if that is the case, don't forget to cite the source please

Reply:
Even though I originally wanted to save time by starting from some pre-existing battle-tested code, I couldn't find anything close to what I needed, so I ended up writing the entire feature from scratch. Zero third-party code. 😄

Reply:
Then it doesn't make sense why doesn't use a single `for-loop`.